src/mango_native_proc.erl

```diff
@@ -283,11 +283,20 @@ get_text_field_type(_) ->
 
 make_text_field(TAcc, Type, Value) ->
     FieldName = make_text_field_name(TAcc#tacc.path, Type),
-    Fields = TAcc#tacc.fields,
-    case Fields == all_fields orelse lists:member(FieldName, Fields) of
+    Fields = case TAcc#tacc.fields of
+        Fields0 when is_list(Fields0) ->
+            % for field names such as "name\\.first"
+            PFields = [mango_doc:parse_field(F) || F <- Fields0],
+            [iolist_to_binary(mango_util:join(".", P)) || {ok, P} <- PFields];
```

Comment:
I see the logic flaw here. It was my attempt to take into account the instance where the user decides to index a field name such as "name\\\\.first". So in this case, we should only index "name.first" and not {"name" : {"first": "some value"}. However, make_text_field_name does not insert escape sequences for field names in the document. So, "name.first" will not match "name\\\\.first", even though it should because that's what we want. Still need to resolve this.